inc/saiobject.h

```diff
@@ -231,6 +231,20 @@ sai_status_t sai_query_attribute_enum_values_capability(
         _In_ sai_attr_id_t attr_id,
         _Inout_ sai_s32_list_t *enum_values_capability);
 
+/**
+ * @brief Query statistics capability
+ *
+ * @param[in] switch_id SAI Switch object id
+ * @param[in] object_type SAI object type
+ * @param[inout] stats_values_capability List of implemented enum values
+ *
+ * @return #SAI_STATUS_SUCCESS on success, #SAI_STATUS_BUFFER_OVERFLOW if list size insufficient, failure status code on error
+ */
+sai_status_t sai_query_stats_capability(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_object_type_t object_type,
+        _Inout_ sai_s32_list_t *stats_values_capability);
```

Comment:
do we know if the stats support read and clear mode or read mode?

Reply:
thats good point, but anyway if vendor dont support read_and_clear in atomic way is there any approach that would eliminate potential read/clrear error ? also if read_and_clear mode is not supported then stats api should return SAI_STATUS_NOT_SUPPORTED, but to have that to know before api call, we would probably need to have another fieldarray param about supported modes

Reply:
I added additional param, as requested, to query list of implemented stat modes (read, read and clear, and future extensions)
Please recheck